src/with_alloc.rs

```diff
@@ -269,10 +264,12 @@ impl<T> AllocRingBuffer<T, PowerOfTwo> {
         unsafe { Self::with_capacity_unchecked(cap) }
     }
 
-    /// Creates an `AllocRingBuffer` with a capacity of [`RINGBUFFER_DEFAULT_CAPACITY`].
+    /// Alias of [`with_capacity`](AllocRingBuffer::with_capacity). Most people will look for a
+    /// new function to create a ringbuffer, but it does not make sense to construct one without
+    /// a capacity.
```

Comment:
This description seems more aimed at the maintainers than users of the crate. Also would we like to mark this as deprecated: https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-deprecated-attribute ?

Reply:
Deprecated for sure, but I stand behind that comment. I did think about it but I think it's nice to mention that an inherent property of the alloc buffer is to have a capacity, and that it's clearer to use with capacity. Saying it like that, rewriting the comment might be nice to make that point even clearer